docs: improve branchConcurrentLimit documentation (#11879)
renovatebot/renovate: docs/usage/configuration-options.md

## branchConcurrentLimit

By default, Renovate won't enforce any concurrent branch limits.

> The `config:base` preset that many extend from limits the amount of concurrent branches.

Reviewer: ```suggestion
The `config:base` preset that many extend from limits the amount of concurrent branches to 20, but in most cases we would recommend lower values such as 5 or 10.
```

}
```

> Warning: increasing the `branchConcurrentLimit` increases the time it takes for Renovate to complete rebasing a PR branch when it's behind the target branch.

Reviewer: ```suggestion
Warning: Leaving PRs/branches as unlimited or as a high number increases the time it takes for Renovate to process a repository.
```

If you find that Renovate is too slow when rebasing out-of-date branches, decrease the `branchConcurrentLimit`.

If you have too many concurrent branches which rebase themselves each run, Renovate can take a lot of time to rebase.
Solutions:

- Decrease the concurrent branch limit (note: this won't go and delete any existing, so won't have an effect until you either merge or close existing ones manually)

> - Remove automerge and/or automatic rebasing (set `rebaseWhen` to `conflicted`). However if you have branch protection saying PRs must be up to date then it's not ideal to remove automatic rebasing.

Author: Not sure if we need this advice or not, I've basically copy/pasted the comment from the maintainer into here. But maybe that's too much information dump?

Maybe we could add this in the FAQ in a section like "Help Renovate rebases are really slow" instead?
Or maybe we don't need this explained at all....

Reviewer: it's ok to leave

> - Remove automerge and/or automatic rebasing (set `rebaseWhen` to `conflicted`). However if you have branch protection saying PRs must be up to date then it's not ideal to remove automatic rebasing.

Author: ```suggestion
- Remove automerge and/or automatic rebasing (set `rebaseWhen` to `conflicted`). However if you have branch protection saying PRs must be up to date then it's not ideal to remove automatic rebasing
```
